app/helpers/dashboards_helper.rb

```diff
@@ -12,4 +12,8 @@ def latest_upload_title(upload)
     return 'Missing required upload' if CSV_TYPES_REQUIRED_TABLE_NAMES.include?(upload.csv_type)
     'Missing upload'
   end
+
+  def can_generate_preview(preview_versions)
+    'disabled' unless preview_versions.none? || !preview_versions[0].generating?
```

Comment:
How about this (easier to read):
```suggestion
    'disabled' if preview_versions[0]&.generating?
```

Reply:
changed